### ISSUE TITLE
Vue Component cannot be passed to mount

### ISSUE THREAD

**Issue by No9** [vuejs/vue-test-utils#636]
### Version
1.0.0-beta.16

### Reproduction link
[https://github.com/No9/vue-test-utils-issue](https://github.com/No9/vue-test-utils-issue)

### Steps to reproduce
See repo
```
> git clone https://github.com/No9/vue-test-utils-issue.git
> cd vue-test-utils-issue
> npm install
> npm test
```

### What is expected?
Passing a Vue Component to `vue-test-utils.mount` should work as per
https://github.com/vuejs/vue-test-utils/issues/329#issuecomment-355501579

### What is actually happening?
An error is thrown that wrapper.html() is undefined

---
I may have misunderstood the use of the wrapper and the library here and unpacking the vue as per the first passing test is expected behavior.

**Comment by eddyerburgh** [comment]
Thanks for the detailed bug report. It was easy to find the problem using your minimal reproduction 😀

The issue was that we weren't compiling extended components correctly. I've fixed this and it will be in the next release. Worth noting that this isn't fixed in Vue < 2.3 due to a bug with mergeOptions.

**Comment by No9** [comment]
Thanks @eddyerburgh - I am on vue@latest so that doesn't impact - 
Glad it was a useful report and thanks again for turning it around.
